fvm/fvm.go

```diff
@@ -75,15 +75,13 @@ func (vm *VirtualMachine) RunV2(
 		WithComputationLimit(uint(proc.ComputationLimit(ctx))).
 		WithMemoryLimit(proc.MemoryLimit(ctx))
 
-	meterParams, err := getEnvironmentMeterParameters(
-		vm,
-		ctx,
-		v,
-		blockPrograms,
-		meterParams,
-	)
-	if err != nil {
-		return fmt.Errorf("error gettng environment meter parameters: %w", err)
+	if ctx.MeterParameters != nil {
+		meterParams = meterParams.
+			WithComputationWeights(ctx.MeterParameters.ComputationWeights()).
+			WithMemoryWeights(ctx.MeterParameters.MemoryWeights())
+		if ctx.MeterParameters.TotalMemoryLimit() != math.MaxUint64 {
+			meterParams = meterParams.WithMemoryLimit(ctx.MeterParameters.TotalMemoryLimit())
+		}
```

Comment:
This shouldn't need to be here.

since context has MeterParameters, there should be a `proc.MeterParameters(ctx)` that internally also does the equivalent of:
```
DefaultParameters().
		WithComputationLimit(uint(proc.ComputationLimit(ctx))).
		WithMemoryLimit(proc.MemoryLimit(ctx))
```